### PR TITLE
Make symbolizer read-time

### PR DESCRIPTION
**Pull request by brancz**
This improves the symbolization story in various ways:

* the metastore is now ephemeral, it can always be recreated when wiped
* instead of always trying to symbolize everything, this only symbolizes at read-time for the data that is actually being queried, and will cache any symbolization results

This overall results in significantly less resources used for symbolization while delivering the same experience as before.

### CODE REVIEW

File: pkg/parca/parca.go [hunk 1]
```diff
@@ -362,9 +378,13 @@ func Run(ctx context.Context, logger log.Logger, reg *prometheus.Registry, flags
 			query.WithTracer(tracerProvider.Tracer("query-engine")),
 		),
 		"stacktraces",
-		parcacol.NewProfileSymbolizer(
-			tracerProvider.Tracer("profile-symbolizer"),
-			mc,
+		symbolizer.New(
+			logger,
+			debuginfoMetadata,
+			symbolizer.NewBadgerCache(db),
```
[the remark this answers is not in the document]
Comment (metalmatze) [reply pkg/parca/parca.go:384]: I guess the PR description answers my questions... :sweat_smile:

File: pkg/profile/schema.go [hunk 2]
```diff
@@ -116,11 +116,11 @@ func SchemaDefinition() *schemapb.Schema {
 			}, {
 				Name: ColumnStacktrace,
 				StorageLayout: &schemapb.StorageLayout{
-					Type: schemapb.StorageLayout_TYPE_STRING,
-					// NOTE: using RLE dictionary causes compaction to explode
-					// in memory due to high cardinality of the dictionaries.
-					Encoding:    schemapb.StorageLayout_ENCODING_PLAIN_UNSPECIFIED,
+					Type:        schemapb.StorageLayout_TYPE_STRING,
+					Encoding:    schemapb.StorageLayout_ENCODING_RLE_DICTIONARY,
```
[the remark this answers is not in the document]
Comment (brancz) [reply pkg/profile/schema.go:120]: correct

File: pkg/parca/parca.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (brancz) [reply pkg/parca/parca.go:384]: That's right, we could also just have an in-memory version of this for example, but I think an on-disk cache is more appropriate for this, as the individual symbolization executions are still fairly expensive, so we still want to do our best to not perform the same symbolization request multiple times.

[the remark this answers is not in the document]
Comment (metalmatze) [reply pkg/parca/parca.go:384]: Makes sense. I guess we can make it configurable if need be at some point.

**Pull request merged**